homeassistant/helpers/aiohttp_compat.py

```diff
@@ -23,3 +23,16 @@ def restore_original_aiohttp_cancel_behavior() -> None:
     """
     web_protocol.RequestHandler = CancelOnDisconnectRequestHandler  # type: ignore[misc]
     web_server.RequestHandler = CancelOnDisconnectRequestHandler  # type: ignore[misc]
+
+
+def enable_compression(response: web.Response) -> None:
+    """Enable compression on the response."""
+    #
+    # Set _zlib_executor_size in the constructor once support for
+    # aiohttp < 3.9.0 is dropped
+    #
+    # We want large zlib payloads to be compressed in the executor
+    # to avoid blocking the event loop, 5192 bytes was chosen because
+    # its also the value used by python-zlib-ng to release the gil
+    response._zlib_executor_size = 5192  # pylint: disable=protected-access
```

Comment:
This statement has no effect on < 3.9.0

Once we drop < 3.9.0 we can set it in the constructor instead